wgsl/index.bs

```diff
@@ -2437,13 +2717,302 @@ The result type depends on the number of letters provided. Assuming a `vec4<f32>
+  <tr algorithm="three component vector selection using .x .y">
+       <td class="nowrap">
+          |e| : vec|N|&lt;|T|&gt;<br>
+          |I| is the letter `x`, `y`, `z`, or `w`<br>
+          |J| is the letter `x`, `y`, `z`, or `w`<br>
+          |K| is the letter `x`, `y`, `z`, or `w`<br>
+       <td class="nowrap">
+           |e|`.`|I||J||K| : vec3&lt;|T|&gt;<br>
+       <td>Computes the three-element vector with first component |e|.|I|, second component |e|.|J|, and third component |e|.|K|.<br>
+           Letter `z` is valid only when |N| is 3 or 4.<br>
+           Letter `w` is valid only when |N| is 4.<br>
+           (OpVectorShuffle)
+  <tr algorithm="three component vector selection using .r .g">
+       <td class="nowrap">
+          |e| : vec|N|&lt;|T|&gt;<br>
+          |I| is the letter `r`, `g`, `b`, or `a`<br>
+          |J| is the letter `r`, `g`, `b`, or `a`<br>
+          |K| is the letter `r`, `g`, `b`, or `a`<br>
+       <td class="nowrap">
+           |e|`.`|I||J||K| : vec3&lt;|T|&gt;<br>
+       <td>Computes the three-element vector with first component |e|.|I|, second component |e|.|J|, and third component |e|.|K|.<br>
+           Letter `b` is only valid when |N| is 3 or 4.<br>
+           Letter `a` is only valid when |N| is 4.<br>
+           (OpVectorShuffle)
+  <tr algorithm="four component vector selection using .x .y">
+       <td class="nowrap">
+          |e| : vec|N|&lt;|T|&gt;<br>
+          |I| is the letter `x`, `y`, `z`, or `w`<br>
+          |J| is the letter `x`, `y`, `z`, or `w`<br>
+          |K| is the letter `x`, `y`, `z`, or `w`<br>
+          |L| is the letter `x`, `y`, `z`, or `w`<br>
+       <td class="nowrap">
+           |e|`.`|I||J||K||L| : vec4&lt;|T|&gt;<br>
+       <td>Computes the four-element vector with first component |e|.|I|, second component |e|.|J|, third component |e|.|K|, and fourth component |e|.|L|.<br>
+           Letter `z` is valid only when |N| is 3 or 4.<br>
+           Letter `w` is valid only when |N| is 4.<br>
+           (OpVectorShuffle)
+  <tr algorithm="four component vector selection using .r .g">
+       <td class="nowrap">
+          |e| : vec|N|&lt;|T|&gt;<br>
+          |I| is the letter `r`, `g`, `b`, or `a`<br>
+          |J| is the letter `r`, `g`, `b`, or `a`<br>
+          |K| is the letter `r`, `g`, `b`, or `a`<br>
+          |L| is the letter `r`, `g`, `b`, or `a`<br>
+       <td class="nowrap">
+           |e|`.`|I||J||K||L| : vec4&lt;|T|&gt;<br>
+       <td>Computes the four-element vector with first component |e|.|I|, second component |e|.|J|, third component |e|.|K|, and fourth component |e|.|L|.<br>
+           Letter `b` is only valid when |N| is 3 or 4.<br>
+           Letter `a` is only valid when |N| is 4.<br>
+           (OpVectorShuffle)
+</table>
+
+#### Component pointer from vector pointer #### {#component-pointer-from-vector-pointer}
+
+<table class='data'>
+  <caption>Getting a pointer to a component from a pointer to a vector</caption>
```

Comment:
MSL doesn't support this.

Closest I could find in the MSL reference is:

> • A pointer or reference to a vector with swizzles is an error:
float4 pos = float4(1.0f, 2.0f, 3.0f, 4.0f);
my_func(&pos.xy); // This is an illegal pointer to a swizzle. 

But that's a multi-letter swizzle.

OpenCL C is more direct about the single-letter case:

> It is illegal to take the address of a vector element and will result in a compilation error. For example:
> float8 vf;
> float *f = &vf.x;          // is illegal
> float2 *f2 = &vf.s07;       // is illegal